### PR TITLE
Try unpinning pytest{-xdist}

### PR DESCRIPTION
**Pull request by dstansby**
xref https://github.com/sunpy/sunpy/issues/6505. pytest 7.2.0 seems to be working locally for me, so lets see what CI has to say about that.

### CODE REVIEW

**Comment by dstansby** [comment]
Looks like we're running into https://github.com/pytest-dev/pytest-cov/issues/557

File: tox.ini [hunk 1]
```diff
@@ -11,7 +11,6 @@ requires =
 isolated_build = true
 
 [testenv]
-pypi_filter = file://.test_package_pins.txt
```
Comment (nabobalis) [inline tox.ini:14]: Don't we want this?

File: .test_package_pins.txt [hunk 2]
```diff
@@ -1,2 +0,0 @@
-pytest<7.2
```
Comment (nabobalis) [inline .test_package_pins.txt:1]: Don't we want to keep this file?

**Comment by dstansby** [comment]
I figured since there was nothing in it the test_package_pins could go - there's no comment in it about what it does, so if it's going to stay (and be empty) could you suggest a comment to put in it describing its purpose?

**Comment by Cadair** [comment]
The file is used by `tox-pypi-filter` as a place for us to exclude packages from test builds for all the sponsored packages. (i.e. [ndcube](https://github.com/sunpy/ndcube/blob/main/tox.ini#L13)) so we shouldn't remove the file even if it's empty as that will cause the link to stop working, and also we want to keep the config in the tox file to make it nice and easy to exclude packages in the future.

Hopefully that provides enough context to document it :see_no_evil:

**Comment by dstansby** [comment]
👍 put back with comment

**Review by nabobalis — APPROVED** [review]

**Comment by nabobalis** [comment]
I wonder if we will need to manually backport something for 4.0 later on.

**Comment by dstansby** [comment]
@MeeseeksDev backport to 4.0

**Pull request merged**